java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1116,6 +1116,30 @@ public boolean isDirectoryEmpty(final String path) throws IOException {
     public boolean existsAndIsAccessible(final String path) {
         try {
             head(path);
+        } catch (MantaClientHttpResponseException e) {
+                if (e.getServerCode().equals(MantaErrorCode.DIRECTORY_DOES_NOT_EXIST_ERROR)) {
```

Comment:
This is a nit so feel free to take it or leave, but with an `if/else` block this large it'd probably be a bit cleaner as a `switch` statement.

Reply:
Refactored into `switch` statement.